scripts/clean-up-localizations.js

```diff
@@ -0,0 +1,65 @@
+const fs = require('fs');
+const glob = require('glob');
+
+const localizationFile_en = 'src/assets/i18n/en_US.json';
+const localizationFile_fr = 'src/assets/i18n/fr_FR.json';
+const localizationFile_de = 'src/assets/i18n/de_DE.json';
+
+// regular expressions for patterns of not explicit used localization keys (dynamic created keys, error keys from rest calls)
+// new patterns have to be added here
+const patterns = ['account.login..*.message', '.*.?error..*'];
+const regEx = new RegExp(patterns.join('|'), 'i');
+
+var filesToBeSearched = [];
+var localizationsFound = {};
+var localizationsFoundOrdered = {};
+
+// store localizations from default localization file in an object
+var localizations = JSON.parse(fs.readFileSync(localizationFile_en, 'utf8'));
+
+var localizations_fr = JSON.parse(fs.readFileSync(localizationFile_fr, 'utf8'));
+var localizations_de = JSON.parse(fs.readFileSync(localizationFile_de, 'utf8'));
+
+// go through directory recursively and find files to be searched
+glob.sync('{src,projects}/**/!(*.spec).{ts,html}').forEach(filePath => {
+  filesToBeSearched.push(filePath);
+});
+
+// add not explicit used localization keys
+Object.keys(localizations)
+  .filter(v => v.match(regEx))
+  .map(localizationKey => {
+    localizationsFound[localizationKey] = localizations[localizationKey];
+    delete localizations[localizationKey];
+  });
+
+filesToBeSearched.forEach(filePath => {
+  const fileContent = fs.readFileSync(filePath);
+  for (var localizationKey in localizations) {
+    if (fileContent.indexOf(localizationKey) > -1) {
+      // store found localizations
+      localizationsFound[localizationKey] = localizations[localizationKey];
+      delete localizations[localizationKey];
+    }
+  }
+});
+
+// sort found localizations
+Object.keys(localizationsFound)
+  .sort()
+  .forEach(function (key) {
+    localizationsFoundOrdered[key] = localizationsFound[key];
+  });
+
+// write found localizations into file
+fs.writeFileSync(localizationFile_en, JSON.stringify(localizationsFoundOrdered, null, 2));
+
+// create cleaned localization files for other languages
+Object.keys(localizationsFoundOrdered).forEach(localizationKey => {
+  localizationsFoundOrdered[localizationKey] = localizations_fr[localizationKey];
+});
+fs.writeFileSync(localizationFile_fr, JSON.stringify(localizationsFoundOrdered, null, 2));
+Object.keys(localizationsFoundOrdered).forEach(localizationKey => {
+  localizationsFoundOrdered[localizationKey] = localizations_de[localizationKey];
+});
```

Comment:
Consider handling `localizationsFoundOrdered` as a `string[]`, then you can remove some of the `Object.keys` invocations and instead just iterate over it.

Reply:
With the use of lodash here this comment is done.